src/shared/watchdog.c

```diff
@@ -114,9 +117,37 @@ int watchdog_set_timeout(usec_t *usec) {
         return r;
 }
 
+usec_t watchdog_runtime_wait(void) {
+        usec_t rtwait = USEC_INFINITY;
+        usec_t ntime;
+
+        if (timestamp_is_set(watchdog_timeout)) {
+                if (timestamp_is_set(watchdog_last_ping)) {
+                        ntime = now(clock_boottime_or_monotonic());
+                        assert(ntime >= watchdog_last_ping);
+                        rtwait = watchdog_last_ping + (watchdog_timeout / 2) - ntime;
+                } else
+                        rtwait = watchdog_timeout / 2;
+
+                if (rtwait <= 0)
+                        return 1;
```

Comment:
why return 1 here, should be zero... But doesn't work anyway, because usec_t is unsigned, see above

Reply:
Previously in manager_loop() wait_usec was set to 1 if wait_usec <= 0, so I kept this here also. I removed it now so we return 0 if the current time is after our target ping window. The manager loop has a rate limit set for looping too fast just in case the ping keeps failing (watchdog device issue). If we reach this rate limit we did anyway our best to ping the watchdog.